src/webapi/mutation_observer.rs

```diff
@@ -0,0 +1,259 @@
+            } );
+        }
+    }
+
+    /// Stops observing all targets.
+    ///
+    /// Until the [`observe`](struct.MutationObserver.html#method.observe) method is called again,
+    /// the `MutationObserver`'s callback will not be called.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver#disconnect())
+    pub fn disconnect( &self ) {
+        js! { @(no_return)
+            @{self.as_ref()}.disconnect();
+        }
+    }
+
+    /// Empties the `MutationObserver`'s record queue and returns what was in there.
+    ///
+    /// This method is generally not needed, instead use the `MutationObserver` callback to respond to changes.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver#takeRecords())
+    pub fn take_records( &self ) -> Vec< MutationRecord > {
+        js!(
+            return @{self.as_ref()}.takeRecords();
+        ).try_into().unwrap()
+    }
+}
+
+
+/// Contains information about an individual change to the DOM.
+///
+/// It is passed to [`MutationObserver`](struct.MutationObserver.html)'s callback.
+///
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/MutationRecord)
+#[ derive( Debug, Clone ) ]
+pub enum MutationRecord {
+    /// One of the target's attributes was changed.
+    Attribute {
+        /// The [`Node`](struct.Node.html) whose attribute changed.
+        target: Node,
+
+        /// The name of the changed attribute.
+        name: String,
+
+        /// The namespace of the changed attribute.
+        namespace: Option< String >,
+
+        /// The value of the changed attribute before the change.
+        old_value: Option< String >,
+    },
+
+    /// The target's data was changed.
+    CharacterData {
+        /// The `CharacterData` node whose data changed.
+        target: Node,
+
+        /// The data of the target before the change.
+        old_data: Option< String >,
+    },
+
+    /// The children of the target were changed.
+    ChildList {
+        /// The [`Node`](struct.Node.html) whose children changed.
+        target: Node,
+
+        /// The nodes which were inserted. Will be an empty [`NodeList`](struct.NodeList.html) if no nodes were inserted.
+        inserted_nodes: NodeList,
+
+        /// The nodes which were removed. Will be an empty [`NodeList`](struct.NodeList.html) if no nodes were removed.
+        removed_nodes: NodeList,
+
+        /// The previous sibling of the inserted or removed nodes, or `None`.
+        previous_sibling: Option< Node >,
+
+        /// The next sibling of the inserted or removed nodes, or `None`.
+        next_sibling: Option< Node >,
+    },
+}
+
+
+impl TryFrom< Value > for MutationRecord {
+    type Error = ConversionError;
+
+    fn try_from( v: Value ) -> Result< Self, Self::Error > {
+        match v {
+            Value::Reference( ref r ) => {
+                let _type: String = js!( return @{r}.type ).try_into()?;
```

Comment:
We usually use the name `kind`. The `_` prefix in Rust has a special meaning - it makes the compiler effectively ignore the variable when generating warnings, and we don't want that here.

Reply:
Ah, good catch! It seems I'm still too used to JavaScript, where using `_` prefix is the norm.